api/v1/views/states.py

```diff
@@ -1,28 +1,57 @@
 #!/usr/bin/python3
 """API endpoint"""
-from flask import abort, jsonify
+from flask import abort, jsonify, request
 from api.v1.views import app_views
 from models import storage
+from models.state import State
 
 @app_views.route('/states', strict_slashes=False)
 def all_states():
     """Return list of all states"""
     all_states = storage.all("State")
     return jsonify([ obj.to_dict() for obj in all_states.values() ])
 
-@app_views.route('/states/<id>', strict_slashes=False)
-def state_by_id(id):
-    """Return State object based off id else raise 404"""
-    state = storage.get("State", id)
-    return jsonify(state.to_dict()) if state else abort(404)
-
-@app_views.route('/states/<id>', strict_slashes=False, methods=['DELETE'])
-def delete_state(id):
-    """Return State object based off id else raise 404"""
-    state = storage.get("State", id)
-    if state:
-        storage.delete()
-        storage.save()
-        return jsonify({}), 200
-    else:
+@app_views.route('/states', strict_slashes=False, methods=['POST'])
+def add_state():
+    """Add state to states"""
+    data = request.get_json()
+    name = data.get('name', None)
```

Comment:
Won't this generate an attribute error if `get_json` returns `None` since `None` doesn't have a `get` method? You could just move the `if not data` between the two above lines.

Reply:
true!